framework/Steps.py

```diff
@@ -552,6 +552,16 @@ def _localInitializeStep(self,inDictionary):
       @ Out, None
     """
     SingleRun._localInitializeStep(self,inDictionary)
+    # check that no input data objects are also used as outputs?
+    for out in inDictionary['Output']:
+      if out.type not in ['PointSet','HistorySet','DataSet']:
+        continue
+      for inp in inDictionary['Input']:
+        if inp.type not in ['PointSet','HistorySet','DataSet']:
+          continue
+        if inp == out:
+          self.raiseAnError(IOError,'The same data object should not be used as both <Input> and <Output> in the same MultiRun step! ' \
+              + 'Step: "{}", DataObject: "{}"'.format(self.name,out.name))
```

Comment:
@alfoa @mandd Are you ok with this changes?

Reply:
is there a reason we need this?

Reply:
Yes, we do need to do this.  

Even with restart sampling, this is the only way to ensure a dataobject has self-consistent sampling data.  I did discuss it with Andrea, and he agreed.  It only affected 4 tests in the repo, none of which actually needed to have the same DO as input and output in the same step.